api/iaas.go

```diff
@@ -148,23 +150,27 @@ func templateCreate(w http.ResponseWriter, r *http.Request, token auth.Token) (e
 	if !allowed {
 		return permission.ErrUnauthorized
 	}
-	evt, err := event.New(&event.Opts{
```

Comment:
I think we could create the event before checking for the template existence. I think it's good to have this failure recorded on the events.